### PR TITLE
DLC close messagetype

### PR DESCRIPTION
**Pull request by dillionverma**
Implementation of DLC close based on discussion in specs.

https://github.com/discreetlogcontracts/dlcspecs/issues/161
https://github.com/discreetlogcontracts/dlcspecs/pull/170

### CODE REVIEW

File: packages/messaging/lib/MessageType.ts [hunk 1]
```diff
@@ -48,6 +48,8 @@ export enum MessageType {
 
   DlcCancelV0 = 52172,
 
+  DlcCloseV0 = 11111, // TODO
```
Comment (dillionverma) [inline packages/messaging/lib/MessageType.ts:51]: There is some ongoing but un-decided discussion here
https://github.com/discreetlogcontracts/dlcspecs/pull/170/files#r674408145

File: packages/messaging/lib/messages/DlcClose.ts [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+
+  public abstract toJSON(): IDlcCloseV0JSON;
+
+  public abstract serialize(): Buffer;
+}
+
+/**
+ * DlcOffer message contains information about a node and indicates its
+ * desire to enter into a new contract. This is the first step toward
+ * creating the funding transaction and CETs.
```
Comment (zpv) [inline packages/messaging/lib/messages/DlcClose.ts:34]: comment needs update

Comment (dillionverma) [reply packages/messaging/lib/messages/DlcClose.ts]: good catch

**Review by zpv — APPROVED** [review]
LGTM 👌

**Review by matthewjablack — APPROVED** [review]

**Pull request merged**